src/VisualStudio/Core/Def/Storage/VisualStudioCloudCacheStorageService.cs

```diff
@@ -17,36 +17,25 @@ namespace Microsoft.VisualStudio.LanguageServices.Storage
     internal class VisualStudioCloudCacheStorageService : AbstractCloudCachePersistentStorageService
     {
         private readonly IAsyncServiceProvider _serviceProvider;
-        private readonly IThreadingContext _threadingContext;
 
-        public VisualStudioCloudCacheStorageService(IAsyncServiceProvider serviceProvider, IThreadingContext threadingContext, IPersistentStorageConfiguration configuration)
+        public VisualStudioCloudCacheStorageService(IAsyncServiceProvider serviceProvider, IPersistentStorageConfiguration configuration)
             : base(configuration)
         {
             _serviceProvider = serviceProvider;
-            _threadingContext = threadingContext;
         }
 
-        protected sealed override void DisposeCacheService(ICacheService cacheService)
-        {
-            if (cacheService is IAsyncDisposable asyncDisposable)
-            {
-                _threadingContext.JoinableTaskFactory.Run(
-                    () => asyncDisposable.DisposeAsync().AsTask());
-            }
-            else if (cacheService is IDisposable disposable)
-            {
-                disposable.Dispose();
-            }
-        }
-
-        protected sealed override async ValueTask<ICacheService> CreateCacheServiceAsync(CancellationToken cancellationToken)
+        protected sealed override async ValueTask<ICacheService> CreateCacheServiceAsync(string solutionFolder, CancellationToken cancellationToken)
         {
             var serviceContainer = await _serviceProvider.GetServiceAsync<SVsBrokeredServiceContainer, IBrokeredServiceContainer>().ConfigureAwait(false);
```

Comment:
Where does the switch from `WaitAndGetResult` to JTF.Run go so that this line won't deadlock?

Reply:
There's the original change in SuggestedAction.cs.